go/teams/loader.go

```diff
@@ -1839,22 +1847,39 @@ func (l *TeamLoader) MapTeamAncestors(ctx context.Context, f func(t keybase1.Tea
 		StaleOK: true, // We only use immutable fields.
 	})
 	if err != nil {
-		return err
+		return NewMapAncestorsError(err, initialTeamIdx)
 	}
 	teamChain := TeamSigChainState{inner: team.Chain}
 	if !teamChain.IsSubteam() {
-		return fmt.Errorf("cannot map over parents of a root team: %v", teamID)
+		return NewMapAncestorsError(
+			fmt.Errorf("cannot map over parents of a root team: %v", teamID),
+			initialTeamIdx,
+		)
 	}
 	return l.mapTeamAncestorsHelper(ctx, f, teamID, teamChain.GetParentID(), reason, forceFullReloadOnceToAssert)
 }
 
-func (l *TeamLoader) mapTeamAncestorsHelper(ctx context.Context, f func(t keybase1.TeamSigChainState) error, teamID keybase1.TeamID, ancestorID *keybase1.TeamID, reason string, forceFullReloadOnceToAssert func(t keybase1.TeamSigChainState) bool) (err error) {
+func (l *TeamLoader) mapTeamAncestorsHelper(
+	ctx context.Context,
+	f func(keybase1.TeamSigChainState, keybase1.TeamName) error,
+	teamID keybase1.TeamID,
+	ancestorID *keybase1.TeamID,
+	reason string,
+	forceFullReloadOnceToAssert func(t keybase1.TeamSigChainState) bool,
+) (err error) {
+	i := 0
+
+	defer func() {
+		if err != nil {
+			err = NewMapAncestorsError(err, i)
```

Comment:
not a huge fan of deferrals like that but I'm fine with keeping this

unless you are on the fence, then i'd vote to just do `NewMapAncestorsError` in every return

Reply:
worried about someone adding something and forgetting to put it